### PR TITLE
wip: convert to backend plugin

### PR DESCRIPTION
**Pull request by kylebrandt**

### CODE REVIEW

**Comment by kylebrandt** [comment]
Note: changes behavior to fix #9

**Comment by kylebrandt** [comment]
Remaining Tasks:

- [ ] Annotations to backend (since timeFilter is now backend)
- [ ] Remove unused JS code
- [ ] Make sure vendor/mod is all up to date
- [ ] CirlceCI runs go tests
- [ ] Packaging

**Comment by erikbos** [comment]
Hi, thanks for working on improving the functionality, looking forward to the new features, really appreciated! Do you happen to take a shoot at https://github.com/grafana/azure-data-explorer-datasource/issues/7 as well 😀  ?

File: README.md [hunk 1]
```diff
@@ -158,44 +158,75 @@ If the command succeeds you should get a result like this:
 
 Queries are written in the new [Kusto Query Language](https://kusdoc2.azurewebsites.net/docs/query/index.html).
```
Comment (erikbos) [inline README.md]: This URL is not fully public, a better documention URL is probably https://docs.microsoft.com/en-us/azure/kusto/query/

File: pkg/azuredx/azuredx.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	err := json.NewEncoder(&buf).Encode(payload)
+	if err != nil {
+		return nil, "", err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, c.ClusterURL+"/v1/rest/query", &buf)
+	if err != nil {
+		return nil, "", err
+	}
+	req.Header.Set("Accept", "application/json")
```
Comment (erikbos) [inline pkg/azuredx/azuredx.go]: According to https://docs.microsoft.com/en-us/azure/kusto/api/rest/request#request-headers setting Host header is mandatory.

What about adding
- req.Header.Set("Accept-Encoding", "gzip, deflate") to get gzip'ed encoded payload (there rest of the function needs to handle that as well)
- req.Header.Set("x-ms-client-request-id", "KGC.execute;"+uuid.Must(uuid.NewV4()).String
()) to set a unique identifier for each request (exactly like the other Kusto client libraries do)

Comment (shavonn) [reply pkg/azuredx/azuredx.go]: Will open a separate issue for gzip header. Added uuid. Thanks!

**Pull request merged**